### PR TITLE
Feat/pendle yt

### PR DESCRIPTION
**Pull request by BowTiedOriole**
I am creating this now for visibility. It is not quite ready as my auto-formatter made a bunch of changes that I need to go back and fix and then do my final review.

I just like to be able to look at it in git when I do my personal final reviews.

### CODE REVIEW

**Comment by BowTiedOriole** [comment]
@coreycaplan3 Sorry for the linting issues. I have fixed that and figured out that slight math issue that we discussed last week. I was impersonating the vault and depositing USDC which was accruing interest and throwing off the math.

I tried to run yarn coverage but my computer crashed. I'm going to try again, but wanted to comment that the code is done.

File: contracts/external/pendle/PendlePtGLP2024IsolationModeVaultFactory.sol [hunk 1]
```diff
@@ -59,19 +59,19 @@ contract PendlePtGLP2024IsolationModeVaultFactory is
...
+    function ownerSetPendleGLPRegistry(
+        address _pendleGLPRegistry
     )
     external
     override
     onlyDolomiteMarginOwner(msg.sender) {
-        pendlePtGLP2024Registry = IPendlePtGLP2024Registry(_pendlePtGLP2024Registry);
-        emit PendlePtGLP2024RegistrySet(_pendlePtGLP2024Registry);
+        pendleGLPRegistry = IPendleGLPRegistry(_pendleGLPRegistry);
+        emit PendleGLPRegistrySet(_pendleGLPRegistry);
```
Comment (coreycaplan3) [inline contracts/external/pendle/PendlePtGLP2024IsolationModeVaultFactory.sol:74]: Don't rename the event either

File: test/external/pendle/PendleYtGLP2024IsolationModeTokenVaultV1.ts [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    it('should fail when not called by vault owner', async () => {
+      const rewardDeposit = [{ marketId: core.marketIds.weth, depositIntoDolomite: false }];
+      await expectThrow(
+        vault.connect(core.hhUser2).redeemDueInterestAndRewards(true, true, rewardDeposit),
+        `IsolationModeTokenVaultV1: Only owner can call <${core.hhUser2.address.toLowerCase()}>`
+      );
+    });
+  });
+
+  describe('#transferIntoPositionWithOtherToken', () => {
```
Comment (coreycaplan3) [inline test/external/pendle/PendleYtGLP2024IsolationModeTokenVaultV1.ts:218]: Rename to `transferFromPositionWithOtherToken`

Be sure to add the new test cases for the added logic discussed in the corresponding Solidity file

**Review by coreycaplan3 — CHANGES_REQUESTED** [review]
See my inline comments

**Review by coreycaplan3 — APPROVED** [review]

**Review by coreycaplan3 — APPROVED** [review]

**Pull request merged**